xmtp_mls/src/groups/message_history.rs

```diff
@@ -129,37 +197,61 @@ where
             log::error!("error publishing sync group intents: {:?}", err);
         }
 
-        Ok(pin_code)
+        Ok((request_id, pin_code))
     }
 
     pub(crate) async fn send_history_reply(
         &self,
         contents: MessageHistoryReply,
-    ) -> Result<(), GroupError> {
+    ) -> Result<(), MessageHistoryError> {
         // find the sync group
         let conn = self.store().conn()?;
-        let sync_group_id = conn
-            .find_sync_groups()?
-            .pop()
-            .ok_or(GroupError::GroupNotFound)?
-            .id;
-        let sync_group = self.group(sync_group_id)?;
+        let sync_group = self.get_sync_group()?;
+
+        // sync the group
+        Box::pin(sync_group.sync(self)).await?;
+
+        let messages = sync_group.find_messages(
+            Some(GroupMessageKind::Application),
+            None,
+            None,
+            None,
+            None,
+        )?;
+
+        let last_message = match messages.last() {
```

Comment:
Nesting match statements can get messy & make it difficult to intuit what's happening


We can compose the `Result<>` and `Option<>` in order to retain one level of indentation

But, we also need the result of the deserialization `from_slice` operation



So, before matching we can map the option:

```rust
let last_message = messages.last().map(|m| serde_json::from_slice(msg.decrypted_message_bytes)).ok().flatten();
match last_message {
    Some(Ok(MessageHistoryContent::Request(request))) => {
        /* ... */
    }
    Some(Ok(MessageHistoryContent::Reply(_) => {
        /* */
    },
    None => return Err(MessageHistoryError::NoPendingRequest)
}
```

Reply:
after implementing your suggestion, i realized that i need access to the value of `msg` inside of the `match`. any tips on how the accomplish that without a nested `match`?